### PR TITLE
Try a more flexible match if the partner can't be found from the bvr …

### PR DESCRIPTION
**Pull request by nbornand**
…reference

### CODE REVIEW

File: sponsorship_switzerland/models/completion_rules.py [hunk 1]
```diff
@@ -79,9 +80,23 @@ def get_from_partner_ref(self, stmts_vals, st_line):
...
+
+        partner_ref = ref[9:16]
         partner_obj = self.env['res.partner']
-        partner = partner_obj.search(
-            [('ref', '=', str(int(ref[9:16])))])
+        partner = partner_obj.search([('ref', '=', str(int(partner_ref)))])
+        if not partner:
+            # Some bvr reference have a wrong number of leading zeros,
+            # resulting in the partner reference to be offset.
+            flexible_ref_match = re.search(r'^0{,10}([1-9]\d{6})0',
```
Comment (ecino) [inline sponsorship_switzerland/models/completion_rules.py]: Not sure what you are doing here. Is it correct the following understanding:

- Up to ten 0
- One digit
- Six digits
- One 0

If that's correct I don't understand the middle part with the digits? [1-9] and \d are the same, no?

File: sponsorship_switzerland/models/completion_rules.py [hunk 2]
```diff
@@ -79,9 +80,23 @@ def get_from_partner_ref(self, stmts_vals, st_line):
...
+                                           ref)
+            if flexible_ref_match:
+                flexible_ref = flexible_ref_match.group(1)
+                if int(flexible_ref) != int(partner_ref):
```
Comment (ecino) [inline sponsorship_switzerland/models/completion_rules.py:94]: Not sure why this test is needed? Wouldn't it work if they were equal?

File: sponsorship_switzerland/models/completion_rules.py (hunk 1 above)
Comment (nbornand) [reply sponsorship_switzerland/models/completion_rules.py]: [1-9] doesn't include 0. This is useful to separate the leading zeros to the partner id (that cannot start with a 0).

File: sponsorship_switzerland/models/completion_rules.py (hunk 2 above)
Comment (nbornand) [reply sponsorship_switzerland/models/completion_rules.py:94]: If they are equal it means that we have already tries to resolve flexible_ref and don't need to try again.

File: sponsorship_switzerland/models/completion_rules.py (hunk 1 above)
Comment (ecino) [reply sponsorship_switzerland/models/completion_rules.py]: There are some smaller refs as well, would it be more risky to change to this?
```suggestion
            flexible_ref_match = re.search(r'^0{,10}([1-9]\d{1,6})0',
```

File: sponsorship_switzerland/models/completion_rules.py [hunk 3]
```diff
@@ -79,9 +80,23 @@ def get_from_partner_ref(self, stmts_vals, st_line):
...
+
+        partner_ref = ref[9:16]
         partner_obj = self.env['res.partner']
-        partner = partner_obj.search(
-            [('ref', '=', str(int(ref[9:16])))])
+        partner = partner_obj.search([('ref', '=', str(int(partner_ref)))])
+        if not partner:
+            # Some bvr reference have a wrong number of leading zeros,
+            # resulting in the partner reference to be offset.
+            flexible_ref_match = re.search(r'^0{,10}([1-9]\d{1,6})0',
```
Comment (nbornand) [inline sponsorship_switzerland/models/completion_rules.py]:
```suggestion
            flexible_ref_match = re.search(r'^0{,10}([1-9]\d{4,6})0',
```

**Pull request merged**